### DISCUSSION TITLE
Desktop/Mobile sidebars for one user

### DISCUSSION THREAD

**Discussion by thkrmr** [elchininet/custom-sidebar#263] [Q&A] [answered]
Hi! How does one go about being able to provide separate sidebars for just a single user that differ depending on whether that user is using desktop or  mobile?

I wasn't able to figure out how to do `exception` matching for `user` in conjunction(combination) with `device`, it seems that as long it's the same user, `extend_from` gets merged between the two separate device definitions that target Android/Chrome.

**Comment by elchininet** [comment]
Hi @thkrmr,
All the exceptions that match are merged among them, but the last match is the one that rules for the final options. Instead of creating a single exception with multiple matchers (that work as `OR` conditions), you can create different exceptions. One base exception for Android and then different exceptions for different users that take that base and override what you want or add new things taking into account that if those users are not in an Android device, only the options that match their user names will be taken into account. You can post some code of your config and I can help you to achieve your goal.
Regards

**Reply by thkrmr** [reply]
Well that's what I have now, that one exception for device: Android (applies to all users) and one exception that matches the user to target the desktop. But that's not what I want to achieve, I don't want to overwrite for all android users. I'd like to match the same user and differentiate based on the device only for that one user, depending which device platform this user opened HA from I'd like extend from different configs. 
Seems like that's impossible as of now?

**Reply by elchininet** [reply]
>Seems like that's impossible as of now?

Yes, right now, the matchers in one exception work as `OR` conditions (I have plans to introduce a new option to change that behaviour, but it is not ready yet, I'll let you know when it is ready).

But at the moment, you can create a Device exception and after that user exceptions (what you are already doing) but if that means many overrides, then you can create user exceptions and after that a small android exception with the things that are common to both.

**Reply by elchininet** [reply, marked as answer]
Hi @thkrmr,
Check the description of [the last release](https://github.com/elchininet/custom-sidebar/releases/tag/v8.1.0) which contains the documentation for the new option, check it and let me know if you face any issue.
Regards

**Reply by thkrmr** [reply]
@elchininet works like a charm with the AND match. Where is the button to buy you a coffee? :) Thank you!

**Reply by elchininet** [reply]
Glad that it works as expected :)
Thanks!
